File: Exercises/Ex6/returned_answers/NestorMancilla/Part2/Normal_distribution.py
import ROOT

file = ROOT.TFile("data.root", "RECREATE")
tree = ROOT.TTree("tree", "Tree with random numbers")
random_number = ROOT.std.vector('float')()
tree.Branch("random_number", random_number)

gRandom = ROOT.TRandom()
for _ in range(1000):
    random_number.push_back(gRandom.Gaus(0, 1))
    tree.Fill()

file.Write()
file.Close()

file = ROOT.TFile("data.root", "READ")
tree = file.Get("tree")
hist = ROOT.TH1F("hist", "Normally Distributed Random Numbers", 100, -4, 4)

hist.SetFillColor(ROOT.kYellow)
tree.Draw("random_number>>hist")

canvas = ROOT.TCanvas()
hist.Fit("gaus")
canvas.SaveAs("histogram.pdf")
file.Close()

File: Exercises/Ex6/returned_answers/NestorMancilla/Part1/Normal_distribution.cc
#include "TFile.h"
#include "TTree.h"
#include "TRandom.h"

int main() {
    gRandom = new TRandom();

    TFile *file = new TFile("data.root", "RECREATE");

    TTree *tree = new TTree("tree", "Tree with normally distributed random numbers");

    float random_number;

    tree->Branch("random_number", &random_number, "random_number/F");

    for (int i = 0; i < 1000; i++) {
        random_number = gRandom->Gaus(0, 1); // Mean = 0, Sigma = 1
        tree->Fill();
    }

    file->Write();
    file->Close();

    return 0;
}
